src/Metadata/Extractor/XmlResourceExtractor.php

```diff
@@ -450,4 +453,23 @@ private function buildGraphQlOperations(\SimpleXMLElement $resource, array $root
 
         return $data;
     }
+
+    private function buildPersistenceMeans(\SimpleXMLElement $resource): ?PersistenceMeansInterface
+    {
+        $persistenceMeans = $resource->persistenceMeans ?? null;
+        if (!isset($persistenceMeans)) {
+            return null;
+        }
+        if (1 !== \count($persistenceMeans)) {
+            throw new \DomainException('There can only be one configured persistence means. "resources.xsd" needs to be fixed.');
+        }
+        $elasticsearchDocument = $persistenceMeans->elasticsearchDocument ?? null;
+        if (isset($elasticsearchDocument)) {
+            return new ElasticsearchDocument(
+                isset($elasticsearchDocument['index']) ? (string) $elasticsearchDocument['index'] : null,
+                isset($elasticsearchDocument['type']) ? (string) $elasticsearchDocument['type'] : null,
+            );
+        }
+        throw new \DomainException(sprintf('Unsupported "%s" persistence means. "resources.xsd" needs to be fixed.', $persistenceMeans->children()?->getName()));
```

Comment:
we're not using domain exceptions probably `RuntimeException` is a better choice

Reply:
https://github.com/api-platform/core/pull/5297#discussion_r1060349412